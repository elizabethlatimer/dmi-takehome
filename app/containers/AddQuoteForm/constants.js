/*
 *
 * AddQuoteForm constants
 *
 */

export const TEMP_QUOTE = 'app/AddQuoteForm/TEMP_QUOTE';
export const CLEAR_TEMP_QUOTE = 'app/AddQuoteForm/CLEAR_TEMP_QUOTE';
