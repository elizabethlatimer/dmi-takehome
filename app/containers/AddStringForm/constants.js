/*
 *
 * AddStringForm constants
 *
 */

export const POST_QUOTE = 'app/AddStringForm/POST_QUOTE';
