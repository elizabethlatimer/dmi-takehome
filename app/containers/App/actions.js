/*
 *
 * Home actions
 *
 */

import {
  LOAD_QUOTES,
  LOAD_QUOTES_SUCCESS,
  LOAD_QUOTES_ERROR,
  ADD_NEW_QUOTE,
  SEND_QUOTE,
} from './constants';

export function loadQuotes() {
  return {
    type: LOAD_QUOTES,
  };
}

export function quotesLoaded(quotes) {
  return {
    type: LOAD_QUOTES_SUCCESS,
    quotes,
  };
}

export function quoteLoadingError(error) {
  return {
    type: LOAD_QUOTES_ERROR,
    error,
  };
}

export function sendQuote() {
  return {
    type: SEND_QUOTE,
  };
}

export function addNewQuote(quote) {
  return {
    type: ADD_NEW_QUOTE,
    quote,
  };
}
