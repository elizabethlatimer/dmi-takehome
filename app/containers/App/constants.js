/*
 *
 * App constants
 *
 */

export const LOAD_QUOTES = 'app/Home/LOAD_QUOTES';
export const LOAD_QUOTES_SUCCESS = 'app/Home/LOAD_QUOTES_SUCCESS';
export const LOAD_QUOTES_ERROR = 'app/Home/LOAD_QUOTES_ERROR';
export const ADD_NEW_QUOTE = 'app/Home/ADD_NEW_QUOTE';
export const SEND_QUOTE = 'app/Home/SEND_QUOTE';
