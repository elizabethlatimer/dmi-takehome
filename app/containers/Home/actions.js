/*
 *
 * Home actions
 *
 */

import {
  LOAD_QUOTES,
  LOAD_QUOTES_SUCCESS,
  LOAD_QUOTES_ERROR,
} from './constants';

export function loadQuotes() {
  return {
    type: LOAD_QUOTES,
  };
}

export function quotesLoaded(quotes) {
  return {
    type: LOAD_QUOTES_SUCCESS,
    quotes,
  };
}

export function quoteLoadingError(error) {
  return {
    type: LOAD_QUOTES_ERROR,
    error,
  };
}
