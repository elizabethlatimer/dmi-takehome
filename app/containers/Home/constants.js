/*
 *
 * Home constants
 *
 */

export const LOAD_QUOTES = 'app/Home/LOAD_QUOTES';
export const LOAD_QUOTES_SUCCESS = 'app/Home/LOAD_QUOTES_SUCCESS';
export const LOAD_QUOTES_ERROR = 'app/Home/LOAD_QUOTES_ERROR';
