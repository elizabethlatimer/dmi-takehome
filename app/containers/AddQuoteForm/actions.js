/*
 *
 * AddQuoteForm actions
 *
 */

import { TEMP_QUOTE, CLEAR_TEMP_QUOTE } from './constants';

export function setTempQuote(quote) {
  return {
    type: TEMP_QUOTE,
    newQuote: quote,
  };
}

export function clearTempQuote() {
  return {
    type: CLEAR_TEMP_QUOTE,
  };
}
