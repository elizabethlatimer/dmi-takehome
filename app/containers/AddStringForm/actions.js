/*
 *
 * AddStringForm actions
 *
 */

import { POST_QUOTE } from './constants';

export function postQuote() {
  return {
    type: POST_QUOTE,
  };
}
