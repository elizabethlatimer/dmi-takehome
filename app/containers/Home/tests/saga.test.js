/**
 * Test sagas
 */

import { takeLatest, put } from 'redux-saga/effects';
import quotesList, { getQuotes } from '../saga';

import { quotesLoaded, quoteLoadingError } from '../../App/actions';
import { LOAD_QUOTES } from '../../App/constants';

/* eslint-disable redux-saga/yield-effects */

describe('getQuotes Saga', () => {
  let getQuotesGenerator;

  beforeEach(() => {
    getQuotesGenerator = getQuotes();

    const callDescriptor = getQuotesGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the quotesLoaded action if it requests the data successfully', () => {
    const response = [
      {
        quote: 'First quote',
        id: 0,
      },
      {
        quote: 'Second quote',
        id: 1,
      },
    ];
    const putDescriptor = getQuotesGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(quotesLoaded(response)));
  });

  it('should call the quoteLoadingError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getQuotesGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(quoteLoadingError(response)));
  });
});

describe('quotesList Saga', () => {
  const quotesListSaga = quotesList();

  it('should start task to watch for LOAD_QUOTES action', () => {
    const takeLatestDescriptor = quotesListSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(LOAD_QUOTES, getQuotes));
  });
});
