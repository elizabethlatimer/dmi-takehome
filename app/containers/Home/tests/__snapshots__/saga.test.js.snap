// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getQuotes Saga should call the quoteLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://localhost:3000/api",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getQuotes Saga should dispatch the quotesLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://localhost:3000/api",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
